File: Groupe2/TP1/src/binaire.c
#include <stdio.h>

// Méthode simplifiée pour afficher un nombre en binaire
void printBinaryFast(int n) {
    // Afficher le nombre en binaire sur 32 bits
    for (int i = 31; i >= 0; i--) {
        printf("%d", (n >> i) & 1);
    }
    printf("\n");
}

// Méthode demandée par l'exercice
void printBinaryExercice(int n) {
    int bits[32]; // Tableau pour stocker les bits (32 bits pour un entier)
    int i = 0;

    // Gérer le cas où n est nul
    if (n == 0) {
        printf("0\n");
        return;
    }

    // Rajouter les bits dans le tableau
    while (n > 0) {
        bits[i] = n % 2;
        n = n / 2;
        i++;
    }

    // Afficher les bits en ordre inverse car on a rempli le tableau à l'envers (du bit de poids faible au bit de poids fort)
    for (int j = i - 1; j >= 0; j--) {
        printf("%d", bits[j]);
    }

    printf("\n");
}

int main() {
    int nombre;

    printf("Entrez un nombre entier : ");
    scanf("%d", &nombre);

    printf("Affichage du nombre en binaire (méthode de l'exercice) : ");
    printBinaryExercice(nombre);

    printf("Affichage du nombre en binaire (méthode simplifiée) : ");
    printBinaryFast(nombre);

    return 0;
}

File: Groupe2/TP1/src/binaire.py
def afficher_binaire_exercice(n):
    # Gérer le cas où n est nul
    if n == 0:
        print("0")
        return

    bits = []

    # Rajouter les bits dans le tableau
    while n > 0:
        bits.append(n % 2)
        n = n // 2

    # Afficher les bits dans l'ordre inverse
    print("Représentation binaire : ", end="")
    for bit in reversed(bits):
        print(bit, end="")
    print()


def main():
    # Lecture de la valeur de l'utilisateur
    nombre = int(input("Entrez un nombre entier : "))

    # Utilisation de la fonction pour afficher le nombre en binaire avec la méthode de l'exercice
    print("Méthode de l'exercice :")
    afficher_binaire_exercice(nombre)

    # Utilisation de la fonction bin() pour convertir le nombre en binaire
    print("Méthode de Python :")
    print(str(bin(nombre))[2:])


if __name__ == "__main__":
    main()
